cpp/cuproj/cmake/thirdparty/get_proj.cmake

```diff
@@ -0,0 +1,49 @@
+# =============================================================================
+# Copyright (c) 2023, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+# in compliance with the License. You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License
+# is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+# or implied. See the License for the specific language governing permissions and limitations under
+# the License.
+# =============================================================================
+
+# This function finds osgeo/proj and sets any additional necessary environment variables.
+function(find_and_configure_proj VERSION)
+  include(${rapids-cmake-dir}/cpm/find.cmake)
+
+  # Find or install Proj
+  rapids_cpm_find(
+    PROJ ${VERSION}
+    #GLOBAL_TARGETS "${global_targets}"
+    BUILD_EXPORT_SET cuspatial-exports
+    INSTALL_EXPORT_SET cuspatial-exports
+    CPM_ARGS
+    GIT_REPOSITORY https://github.com/osgeo/proj.git
+    GIT_TAG ${VERSION}
+    GIT_SHALLOW TRUE
+    FIND_PACKAGE_ARGUMENTS "${find_package_args}"
+  )
+
+  if(PROJ_ADDED)
+    rapids_export(
+      BUILD PROJ
+      VERSION ${PROJ_VERSION}
+      EXPORT_SET PROJTargets
+      GLOBAL_TARGETS PROJ
+      NAMESPACE PROJ::
+    )
+
+    include("${rapids-cmake-dir}/export/find_package_root.cmake")
+    rapids_export_find_package_root(
+      BUILD Proj [=[${CMAKE_CURRENT_LIST_DIR}]=] cuspatial-testing-exports
+    )
+  endif()
+
+endfunction()
```

Comment:
Do we need to generate a `proj-config.cmake` in our build dir?

Reply:
I don't know anything about what export means in cmake. I don't really understand how cmake commands map to the much simpler underlying concepts like -I and -L command line arguments.

Reply:
Who would be able to answer your question? Shouldn't PROJ provide such a thing?

Reply:
The PROJTargets in the above is wrong, fails in CI.

Reply:
According to the docs, `rapids_export` generates a proj-config.cmake. https://docs.rapids.ai/api/rapids-cmake/nightly/command/rapids_export/